From: amodra <amodra@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Wed, 4 May 2016 13:23:58 +0000 (+0000)
Subject: [RS6000] Correct PIC_OFFSET_TABLE_REGNUM
X-Git-Tag: gcc-7_1_0-release~7184
X-Git-Url: https://gcc.gnu.org/git/?p=gcc.git;a=commitdiff_plain;h=9213244550335bcb2b8590a0d7d58ac74c932361

[RS6000] Correct PIC_OFFSET_TABLE_REGNUM

Leaving this as r30 results in pic_offset_table_rtx of (reg 30)
for -m64, which is completely bogus.  Various rtl analysis predicate
functions treat pic_offset_table_rtx specially..

	* config/rs6000/rs6000.h (PIC_OFFSET_TABLE_REGNUM): Correct.


git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/trunk@235876 138bc75d-0d04-0410-961f-82ee72b054a4
---

gcc/

2016-05-04  Alan Modra  <amodra@gmail.com>

	* config/rs6000/rs6000.h (PIC_OFFSET_TABLE_REGNUM): Correct.
--- a/src/gcc/config/rs6000/rs6000.h
+++ b/src/gcc/config/rs6000/rs6000.h
@@ -2058,7 +2058,10 @@ do {									     \
    to allocate such a register (if necessary).  */
 
 #define RS6000_PIC_OFFSET_TABLE_REGNUM 30
-#define PIC_OFFSET_TABLE_REGNUM (flag_pic ? RS6000_PIC_OFFSET_TABLE_REGNUM : INVALID_REGNUM)
+#define PIC_OFFSET_TABLE_REGNUM \
+  (TARGET_TOC ? TOC_REGISTER			\
+   : flag_pic ? RS6000_PIC_OFFSET_TABLE_REGNUM	\
+   : INVALID_REGNUM)
 
 #define TOC_REGISTER (TARGET_MINIMAL_TOC ? RS6000_PIC_OFFSET_TABLE_REGNUM : 2)
 
